controllers/templatesync/template_sync.go

```diff
@@ -552,7 +675,28 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 
 			successMsg := fmt.Sprintf("Policy template %s was updated successfully", tName)
 
-			err = r.handleSyncSuccess(ctx, instance, tIndex, tName, successMsg, res)
+			// Handle cluster scoped objects
+			if isClusterScoped {
+				addFinalizer = true
+
+				reqLogger.V(2).Info("Finalizer required for " + gvk.Kind)
+
+				// Ensure the CRD is labeled as policy-type=template so we can query custom resources for
+				// cleanup
+				err := applyCRDLabel(ctx, instance, tName, rsrc, dClient)
```

Comment:
Should we only set this when the policy template is Gatekeeper related even if it's cluster-scoped?

Reply:
Right now, the controller watch list is filtered by the policy-template label, so we could restrict it to Gatekeeper-related objects, but I'm wondering now if we should have a policy allowlist and keep this particular check generic:
https://github.com/open-cluster-management-io/governance-policy-framework-addon/blob/1048a8040d8a20bddec82b1afed3f89a26560198/main.go#L339-L351

You're bringing up a good point that now that I've added CRD permissions, there's zero filtering for the template-sync passing along a CRD. (@JustinKuli tested it syncing an Event since it has those permissions--and it synced it.)